WIT.md

```diff
@@ -44,16 +44,41 @@ whitespace ::= ' ' | '\n' | '\r' | '\t'
 
 ### Comments
 
-A `comment` token in `*.wit` is either a line comment preceded with `//` which
-ends at the next newline (`\n`) character or it's a block comment which starts
+Commenting your code might have two purposes: implementation details and documentation. 
+Therefore, there are two types of comments for each purpose. 
+
+#### Implementation comments
+
+An `implementation comment` token in `*.wit` is either a line comment preceded with `//` 
+which ends at the next newline (`\n`) character or it's a block comment which starts
 with `/*` and ends with `*/`. Note that block comments are allowed to be nested
 and their delimiters must be balanced
 
 ```wit
-comment ::= '//' character-that-isnt-a-newline*
-          | '/*' any-unicode-character* '*/'
+implementation-comment ::= '//' character-that-isnt-a-newline*
+                         | '/*' any-unicode-character* '*/'
 ```
 
+#### Documentation comment
+
+A `documentation comment` token in `*.wit` is a documentation line preceded with `///` 
+which ends at the next newline (`\n`) character or it's a block doc comment which starts
+with `/**` and ends with `**/`. Note that block doc comments are allowed to be nested
+and their delimiters must be balanced. Place documentation comments 
+just before the item they’re documenting. 
+
+The text considered for the documentation is constructed as follows.
+For line doc comments, all leading `/` are ignored. Then, leading and trailing whitespaces
+in each line (either in lines or blocks) is also ignored. Note that the number of lines is 
+preserved as given in the `*.wit`: different documentation lines in doc comments are different 
+lines in the final documentation.
+
+```wit
+documentation-comment ::= '///' '/'* character-that-isnt-a-newline*
```

Comment:
I don't think the `'/'*` is necessary here, right? (not sure why there'd be allowed even more slashes explicitly)

Reply:
I wrote this because (currently) immediate `'/'` are trimmed and therefore are not part of the final documentation.

In other words, after `///` any character counts as part of the documentation, except `/`.

The implementation is [this line](https://github.com/bytecodealliance/wit-bindgen/pull/125/files#diff-2164780efe09de34f76096a0bcfdefb5b1f4226ce229361f647c9174335a7961L518-R518)

In Rust, if I am not mistaken, a line starting with `////` is considered a comment and is not part of the documentation. Should I do this?

Reply:
I think that's just a weird artifact of the current implementation but I don't think there's any need to codify it, this just needs to say that doc-comments start with three slashes.